### PR TITLE
New design for phishing warning page

### PR DESCRIPTION
**Pull request by aleksandar-mihajlovic**
## Explanation
Added new design for phishing warning page according to [figma design](https://www.figma.com/file/MKJnCnI8Xi2EVcRCNAHwOu/Improve-Phishing-Warning-Page?node-id=2%3A9&t=P2jLXBxUvqxA3A7d-0).

* Fixes #16602
 
### Before
Screenshot 2023-01-26 at 16 01 52

### After
phishingPageNew

## Manual Testing Steps

Production testing:
1. Go to one of the links: primalfrens.club, geishateahouse.io, cryptocity.fun
2. Phishing warning page is displayed
3. All links are working
4. Advisory provided by is written based on the source of the detection repo (in link there is info if the Metamask or Phishfort is detection repo)
5. When back to safety is clicked, tab is closed

Local testing:
1. static/ folder
2. Double click on index.html
3. Phishing warning page is displayed
4. All links are working expect continue to the site, because production app is needed for that
5. Advisory provided by is written based on the source of the detection repo (in link there is info if the Metamask or Phishfort is detection repo)
6. When back to safety is clicked, tab is closed

### CODE REVIEW

File: static/index.css [hunk 1]
```diff
@@ -32,17 +32,38 @@ body {
   width: 100vw;
   height: 100vh;
 
-  background-color: var(--color-error-default);
+  background-color: #d03242;
```
Comment (georgewrmarshall) [inline static/index.css:35]: As a general rule a static hex value should be a red flag. We should be using the design token css color variables available

```suggestion
  background-color: var(--color-error-default);
```

File: static/index.css [hunk 2]
```diff
@@ -32,17 +32,38 @@ body {
...
 }
 
-p {
-  margin: 2em;
-}
-
 p a {
   text-decoration: underline;
-  color: var(--color-primary-default);
+  color: white;
```
Comment (georgewrmarshall) [inline static/index.css:41]: Can we use design token CSS vars here?

```suggestion
  color: var(--color-error-inverse);
```

File: static/index.css [hunk 3]
```diff
@@ -32,17 +32,38 @@ body {
...
+  text-shadow: 0px 4px rgba(0,0,0, 0.15);
+}
+
+ul {
+  padding-left: 2em;
+}
+
+button {
+  color: red;
+  background: white;
```
Comment (georgewrmarshall) [inline static/index.css:59]: Use color tokens 
```suggestion
  background: var(--color-error-inverse);
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @aleksandar-mihajlovic, nice work matching the Figma design but I think it breaks some of the MetaMask design system conventions that I think should be resolved before this get's merged.

Hey @holantonela, I've left some comments in the Figma file https://www.figma.com/file/MKJnCnI8Xi2EVcRCNAHwOu?node-id=4:49#355806441

File: static/index.css (hunk 1 above)
Comment (aleksandar-mihajlovic) [reply static/index.css:35]: Yes, seen that little late, will fix it

File: static/index.css (hunk 2 above)
Comment (aleksandar-mihajlovic) [reply static/index.css:41]: Yes, seen that little late, will fix it

File: static/index.css (hunk 3 above)
Comment (aleksandar-mihajlovic) [reply static/index.css:59]: Yes, seen that little late, will fix it

**Comment by dzfjz** [comment]
Verified by QA

File: static/index.css (hunk 1 above)
Comment (holantonela) [reply static/index.css:35]: You should use the error default red color @aleksandar-mihajlovic

**Comment by holantonela** [comment]
@georgewrmarshall commented in the figma. Someone introduced that change, but i bet it was a click error. Please, @aleksandar-mihajlovic lets remove the border and the shadow of the title. Thank you!

File: static/index.css (hunk 1 above)
Comment (aleksandar-mihajlovic) [reply static/index.css]: It is fixed now

File: static/index.css [hunk 4]
```diff
@@ -32,17 +32,30 @@ body {
...
+button {
+  color: var(--color-error-alternative);
+  background: var(--color-error-inverse);
+  padding: 1em;
+  border-radius: 2em;
+  border: none;
+  float: right;
+  margin-top: 3em;
+  font-size: 0.9em;
+  font-family: Euclid, Roboto, Helvetica, Arial, sans-serif;
```
Comment (georgewrmarshall) [inline static/index.css:58]: @garrettbear can you please suggest the correct css for the button here?

You can add some of the typography tokens here in favour of the static typography styles @aleksandar-mihajlovic 

```scss
button {
  font-family: var(--typography-s-body-md-font-family);
  font-weight: var(--typography-s-body-md-font-weight);
  font-size: var(--typography-s-body-md-font-size);
  line-height: var(--typography-s-body-md-line-height);
  letter-spacing: var(--typography-s-body-md-letter-spacing);

  @media screen and (min-width: 768px) {
    font-family: var(--typography-l-body-md-font-family);
    font-weight: var(--typography-l-body-md-font-weight);
    font-size: var(--typography-l-body-md-font-size);
    line-height: var(--typography-l-body-md-line-height);
    letter-spacing: var(--typography-l-body-md-letter-spacing);
  }
}
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Have made one more suggestion. Thanks for your patience @aleksandar-mihajlovic 🙏

File: static/index.css (hunk 4 above)
Comment (georgewrmarshall) [inline static/index.css:58]: You can update the typography style to use the design token CSS variables for the typography too.

```suggestion
  color: var(--color-error-alternative);
  background: var(--color-error-inverse);
  padding: 1em;
  border-radius: 2em;
  border: none;
  float: right;
  margin-top: 3em;
  font-family: var(--typography-s-body-md-font-family);
  font-weight: var(--typography-s-body-md-font-weight);
  font-size: var(--typography-s-body-md-font-size);
  line-height: var(--typography-s-body-md-line-height);
  letter-spacing: var(--typography-s-body-md-letter-spacing);

  @media screen and (min-width: 768px) {
    font-family: var(--typography-l-body-md-font-family);
    font-weight: var(--typography-l-body-md-font-weight);
    font-size: var(--typography-l-body-md-font-size);
    line-height: var(--typography-l-body-md-line-height);
    letter-spacing: var(--typography-l-body-md-letter-spacing);
  }
```

File: static/index.html [hunk 5]
```diff
@@ -39,74 +43,40 @@
...
+      </br>
+        <p>Potential threats on <span id="suspect-link"></span> include:
+          <ul>
+            <li>Fake versions of MetaMask</li>
+            <li>Secret Recovery Phrase or password theft</li>
+            <li>Malicious transactions resulting in stolen assets </li>
+          </ul>
         </p>
+      </br>
+        <p>Advisory provided by Ethereum Phishing Detector, CryptoScamDB, and PhishFort.</p>
```
Comment (holantonela) [inline static/index.html:74]: This list should be dynamic. We should expose ONLY the sources who are flagging the current URL @aleksandar-mihajlovic @dzfjz

File: static/index.html [hunk 6]
```diff
@@ -39,74 +43,40 @@
       charset="utf-8"
     ></script>
     <link rel="stylesheet" type="text/css" href="./design-tokens.css">
+    <link rel="stylesheet" href="https://raw.githubusercontent.com/MetaMask/design-tokens/main/src/css/design-tokens.css">
```
Comment (georgewrmarshall) [inline static/index.html:46]: I don't think we want to include this line it was just to check if we could get the phishing page to render with the correct design token css variables. We do need a dev environment for this repo though.

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. There maybe one more update needed to the button that is being resolved in the Figma file

File: static/index.html [hunk 7]
```diff
@@ -11,6 +11,10 @@
         document.getElementById('antiClickjack').innerHTML =
           '#content__framed-body { display: none !important; }';
       }
+
+      window.onload = function() {
+        document.getElementById("suspect-link").innerHTML = document.referrer;
```
Comment (NicholasEllul) [inline static/index.html:16]: Setting `.innerHTML` can have [security implications](https://dev.to/caffiendkitten/innerhtml-cross-site-scripting-agc).  In this case innerText should work as well 👍 

```suggestion
        document.getElementById("suspect-link").innerText = document.referrer;
```

File: static/index.html [hunk 8]
```diff
@@ -39,74 +43,40 @@
...
+            <li>Malicious transactions resulting in stolen assets </li>
+          </ul>
         </p>
+      </br>
+        <p>Advisory provided by Ethereum Phishing Detector, CryptoScamDB, and PhishFort.</p>
+      </br>
+      </br>
+        <p>If we're flagging a legitimate website, please <a id="new-issue-link" href='' onclick="window.open('https://github.com/metamask/eth-phishing-detect/issues/new')">report a detection problem.</a></p>
+        <p>If you understand the risks and still want to proceed, you can <a id="unsafe-continue">continue to the site.</a></p>
+        <button class="button btn--rounded btn-primary" type="submit" onclick="window.close()">Back to safety</button>
```
Comment (NicholasEllul) [inline static/index.html:79]: On mobile, clicking back to safety returns the user to the previous page. Do we want to do `history.back()` here?

**Review by NicholasEllul — COMMENTED** [review]
This page looks fantastic! Glad to see it being implemented

File: static/index.html (hunk 8 above)
Comment (holantonela) [reply static/index.html:79]: We agreed that closing the tab is safer and more reliable than going back. The behaviour should be the same in mobile and desktop. The discussion is here https://www.figma.com/file/MKJnCnI8Xi2EVcRCNAHwOu?node-id=4:49#266099480

File: static/index.html (hunk 5 above)
Comment (holantonela) [reply static/index.html:74]: To do so, we need to add a query string in the `PhishingController` to say which list flagged it. cc @409H

Comment (holantonela) [reply static/index.html:74]: This is the point 2 in the **Acceptance Criteria** 

> 2. Users can read the attribution to who flagged the domain, probably passed to that page with a query string.

https://github.com/MetaMask/metamask-extension/issues/16602

**Comment by aleksandar-mihajlovic** [comment]
@georgewrmarshall can you check the attached gif, is the button okay now? I used the Button Secondary as mentioned in Figma.

File: static/index.html (hunk 5 above)
Comment (Gudahtt) [reply static/index.html:73]: Thanks @holantonela , great catch

>Well, as I understand, these websites are not scam websites, they are websites that give users info about how to avoid phishing, so they don't need to be listed dynamically. Am I wrong?

I can explain the purpose of this a criteria a little more. The concern is that we want the user to know _specifically_ why the site is blocked, so that they can know who to file an issue with if they believe it was flagged in error. For example, we don't want users making issues on our phishing detect repo for sites that are blocked by Phishfort, because we can't do anything about that. They would need to go to Phishfort.

To that end, we'll want to update the "report a problem" link to be dynamic as well, so it points the user in the right direction.

Comment (aleksandar-mihajlovic) [reply static/index.html:73]: Okay, thank you for this clarification, this is was very helpful. I will investigate and fix this. Just one question, are there more detect repo's besides Metamask and Phishfort? @Gudahtt

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! I've created another PR with some more CSS suggestions https://github.com/MetaMask/phishing-warning/pull/55 that you can review and merge into this PR if you think they are acceptable.

**Comment by aleksandar-mihajlovic** [comment]
I have added the CSS code that you suggested @georgewrmarshall if you can check it out.
I have added logic for setting dynamically text and link also, for phishing detection repo that detected phishing website, if you can check it out @holantonela.

File: static/index.html [hunk 9]
```diff
@@ -46,67 +58,32 @@
...
-          >.
-        </p>
-        <p>
-          Note that this warning list is compiled on a voluntary basis. This
-          list may be inaccurate or incomplete. Just because a domain does not
-          appear on this list is not an implicit guarantee of that domain's
-          safety. As always, your transactions are your own responsibility. <span id="redirect-warning-message">If
-          you wish to interact with any domain on our warning list, you can do
-          so by <a id="unsafe-continue">continuing at your own risk</a>.</span>
+          MetaMask flagged the site you're trying to visit as potentially deceptive. Attackers may trick you into doing something dangerous. <a id="csdbLink" href='' onclick="window.open('https://cryptoscamdb.org/search')">Learn more</a>
```
Comment (weizman) [inline static/index.html:70]: this call unsafely allows reverse tab nabbing attacks (read more [here](https://github.com/orgs/MetaMask/projects/39/views/1?filterQuery=nabb&pane=issue&itemId=16010904)) - please fix by adding attribute `rel="noopener"`

**Review by weizman — CHANGES_REQUESTED** [review]
minor security concern

File: static/index.html (hunk 5 above)
Comment (Gudahtt) [reply static/index.html]: Nope, we only support those two at the moment

File: static/index.html (hunk 9 above)
Comment (weizman) [reply static/index.html:70]: also, why use `onclick` and `open` when you have the default behaviour of an anchor element?

`<a id="csdbLink" rel="noopener" href='https://cryptoscamdb.org/search' target='_blank'">Learn more</a>`

Comment (aleksandar-mihajlovic) [reply static/index.html]: Okay no problem, if it is making security problems, i will changed it right now, you can look at it, if it is okay. @weizman

**Comment by AlexHerman1** [comment]
can we make sure that whatever ends up going to prod, we make the attribution prominent and point them towards the correct contributor to raise an issue?

**Review by georgewrmarshall — DISMISSED** [review]
Looks great! Going to approve for visual and design system requirements. Nice work! 💯  Please don't merge until all reviewers have approved 🙏

File: static/index.html (hunk 5 above)
Comment (Gudahtt) [reply static/index.html]: @409H thanks for linking that, I had failed to consider this requirement when reviewing a cost-saving change to how we poll for the phishing list. Unfortunately this information will no longer be available after that change because we're now combining all lists into one to save on network bandwidth.

Still, we can fulfill this requirement without that data by checking on-demand after a user chooses to dispute a block. Tomorrow I will create a separate issue detailing how this needs to change.

Comment (Gudahtt) [reply static/index.html]: @aleksandar-mihajlovic My apologies, I hadn't looked closely at this PR when I left my first comment.

I don't think we need the onload function you added for this, because that same logic is already handled by `index.ts`. It was implemented here: #23. We pass in a custom dispute URL if a site is blocked by Phishfort, otherwise we default to ours. As long as that `newIssueUrl` continues to be set dynamically on this link, no change to this behavior is needed here.

Comment (aleksandar-mihajlovic) [reply static/index.html]: @Gudahtt Okay, i removed that part of the code for linking on the onload function, and returned to the old state like you mentioned, is it okay now?

**Comment by holantonela** [comment]
> can we make sure that whatever ends up going to prod, we make the attribution prominent and point them towards the correct contributor to raise an issue?

Yes @AlexHerman1, we discussed it [here](https://github.com/MetaMask/phishing-warning/pull/52#pullrequestreview-1277832723)

File: src/index.test.ts [hunk 10]
```diff
@@ -71,16 +71,7 @@ describe('Phishing warning page', () => {
 
   it('should render page', async () => {
     const container = window.document.getElementsByTagName('body')[0];
-    expect(
-      queryByText(container, 'MetaMask Phishing Detection'),
-    ).not.toBeNull();
-  });
-
-  it('should have correct default "New issue" link', () => {
```
Comment (Gudahtt) [inline src/index.test.ts:79]: It seems that this test should still work - was there a reason this was removed? It would be ideal to preserve it.

File: src/index.ts [hunk 11]
```diff
@@ -137,16 +137,6 @@ function start() {
     throw new Error('Unable to locate unsafe continue link');
   }
 
-  if (isValidSuspectHref(suspectHref) === false) {
```
Comment (Gudahtt) [inline src/index.ts:140]: Curious why this was removed. This was to prevent redirecting to unsupported protocols. We'll need to continue protecting against this at least as long as we're passing the blocked URL as a query parameter (i.e. so long as it's untrusted input).

File: static/index.html (hunk 5 above)
Comment (Gudahtt) [reply static/index.html]: Thanks @aleksandar-mihajlovic , yes this thread looks resolved now

File: static/index.html [hunk 12]
```diff
@@ -11,6 +11,17 @@
         document.getElementById('antiClickjack').innerHTML =
           '#content__framed-body { display: none !important; }';
       }
+
+      window.onload = function() {
+        document.getElementById("suspect-link").innerText = document.referrer;
```
Comment (Gudahtt) [inline static/index.html:16]: Nit: Could you move this code to the `start` function in `src/index.ts`? That is where all of our other "page initialization" code is.

File: src/index.ts (hunk 11 above)
Comment (Gudahtt) [reply src/index.ts:140]: Oh I see. The affected component was removed in this redesign.

This is not critical then. This is a bizarre edge cases that would only be countered by users redirected to this page by a phisher.

**Review by Gudahtt — APPROVED** [review]
LGTM! I can address the remaining two nits in a follow-up PR.

File: src/index.test.ts (hunk 10 above)
Comment (Gudahtt) [reply src/index.test.ts:79]: Restored here: https://github.com/MetaMask/phishing-warning/pull/60
